Sources/App/Core/AppMetrics.swift

```diff
@@ -105,6 +105,22 @@ enum AppMetrics {
     static var apiSearchGetTotal: PromCounter<Int, EmptyLabels>? {
         counter("spi_api_search_get_total", EmptyLabels.self)
     }
+    
+    static var apiSearchGetWithFilterTotal: PromCounter<Int, EmptyLabels>? {
+        counter("spi_api_search_get_with_filter_total", EmptyLabels.self)
+    }
+    
+    static func apiSearchGetWithFilterTotal(key: String) -> PromCounter<Int, EmptyLabels>? {
+        counter("spi_api_search_get_with_filter_\(key)_total", EmptyLabels.self)
```

Comment:
`key` should be a label, and we should also probably constrain it to an `enum`. This will also make `spi_api_search_get_with_filter_total` superfluous: you can filter by label individually and against the total without labels.

I'll take a look and will open a PR on top of this one.

Reply:
Making it a label in hindsight would've been a smart idea, agreed.

I do get where you're coming from with respect of using an enum for the key, i just wanted to try and make it so that you didn't have to think about metrics every time you add a new filter. With an enum, it would force us to update it each time a new metric is added

Reply:
I've just opened a PR with some minimal changes. I think there's more we could do here to make it type-safe and improve testability/maintenance a bit but that's all minor and not related to the labels themselves.